src/app/infrastructure/TimeRangeUtils.ts

```diff
@@ -94,4 +93,23 @@ export function getDefaultTimePickerRange (dayForTime?: moment.Moment): {startTi
          endTime: getDefaultTimePickerMaxTime(moment())
         };
     }
+}
+
+export function roundTimeToNearestQuaterHour (timeToRound: moment.Moment): moment.Moment {
+    const minutes = moment(timeToRound).minutes();
+    const hour = moment(timeToRound).hours();
+
+    if (minutes <= 7) {
```

Comment:
I wonder if this could be done easier with math.

```
const roundedMinutes = Math.round(minutes / 15.0)*15;
```

This would also give an advantage of allowing the increment to be passed in as a parameter.  Given you have already written your tests tests for this method, it should be easy to try it out.